pkg/util/css.go

```diff
@@ -0,0 +1,122 @@
+package util
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/agext/regexp"
+	"github.com/andybalholm/cascadia"
+	"github.com/pkg/errors"
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+)
+
+func escapeChar(r rune) string {
+	return fmt.Sprintf("\\%x ", int(r))
+}
+
+var nonIdentifier = regexp.MustCompile("[^a-zA-Z0-9_-]")
+var extraSpace = regexp.MustCompile("\\s+")
+
+// Note - this is a rudimentary implementation
+func escapeCSSIdentifier(input string) string {
+	if len(input) == 0 {
+		return ""
+	}
+
+	// Matches CSS non-identifier characters
+	input = nonIdentifier.ReplaceAllStringFunc(input, func(match string) string {
+		return escapeChar([]rune(match)[0])
+	})
+
+	// If identifier starts with a digit, hyphen + digit or two hyphens, escape it
+	firstChar := []rune(input)[0]
+	if firstChar == '-' {
+		if len(input) > 1 && (input[1] == '-' || (input[1] >= '0' && input[1] <= '9')) {
+			input = escapeChar(firstChar) + input[1:]
+		}
+	} else if firstChar >= '0' && firstChar <= '9' {
+		input = escapeChar(firstChar) + input[1:]
+	}
+
+	return input
+}
+
+func getAttr(n *html.Node, key string) string {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return attr.Val
+		}
+	}
+	return ""
+}
+
+// Get a CSS selector that will uniquely select a provided HTML element
+// If the element has an ID, returns #id;
+// otherwise returns the parent (if any) CSS selector, followed by '>'
+// followed by a unique selector for the element (tag.class.class:nth-child(n)).
+// Logic copied from JSoup: https://github.com/jhy/jsoup/blob/0b10d516ed8f907f8fb4acb9a0806137a8988d45/src/main/java/org/jsoup/nodes/Element.java#L829
+func CSSSelector(n *html.Node) string {
```

Comment:
This will become part of the public API surface. Shouldn't this be under `pkg/internal/util`?

Reply:
@mickael-menu you're correct. I've moved it